### PR TITLE
Fix 53648  

### PR DESCRIPTION
**Pull request by Jammer77**
VO notification for "Select All" and "Remove/Add breakpoint"

### CODE REVIEW

File: main/src/addins/MacPlatform/MacPlatform.cs [hunk 1]
```diff
@@ -280,6 +280,9 @@ public override bool SetGlobalMenu (CommandManager commandManager, string comman
 			return true;
 		}
 
+
```
Comment (iainx) [inline main/src/addins/MacPlatform/MacPlatform.cs]: This method is needed to flash the menu, so while removing it might stop the Divider message, it also removes necessary functionality.

File: main/src/addins/MacPlatform/MacPlatform.cs [hunk 2]
```diff
@@ -293,13 +296,15 @@ static void OnCommandActivating (object sender, CommandActivationEventArgs args)
...
+		*/
+		
 
 		void InitApp (CommandManager commandManager)
 		{
 			if (initedApp)
 				return;
 
-			commandManager.CommandActivating += OnCommandActivating;
+			//commandManager.CommandActivating += OnCommandActivating; //that code reason of "divider" message in Voice Over
```
Comment (iainx) [inline main/src/addins/MacPlatform/MacPlatform.cs]: This code is needed as well.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs [hunk 3]
```diff
@@ -225,6 +225,7 @@ public DefaultWorkbench()
 			DockNotebook.NotebookChanged += NotebookPagesChanged;
 
 			Accessible.SetIsMainWindow (true);
+			Accessible.SetRole (AtkCocoa.Roles.AXButton);
```
Comment (iainx) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs]: Why are you making the root window have a button role?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs [hunk 4]
```diff
@@ -1135,6 +1136,12 @@ protected override bool OnFocusInEvent (Gdk.EventFocus evnt)
 			return base.OnFocusInEvent (evnt);
 		}
 
+		protected override void OnSetFocus (Widget focus)
```
Comment (iainx) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs]: What is this method for, it doesn't seem related to the breakpoint code

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs (hunk 3 above)
Comment (Jammer77) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs]: Because i have no choice:

public enum Roles
        {
            AXButton,
            AXCell,
            AXColumn,
            AXGroup,
            AXGrowArea,
            AXImage,
            AXLink,
            AXList,
            AXMenuButton,
            AXPopUpButton,
            AXRadioButton,
            AXRow,
            AXRuler,
            AXSplitGroup,
            AXSplitter,
            AXStaticText,
            AXTabGroup,
            AXTextArea
        };

Comment (iainx) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs]: You can add the missing role to the Roles enum, but why does it need a custom role set? GtkWindows should already have the AXWindow role set

Comment (Jammer77) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs]: Ok.

**Comment by iainx** [comment]
I made some tweaks to the FlashMenu method so that it reads the command name so we don't need to go through and add the code to every command.
